.github/workflows/get_rocks_modified_and_build_scan_test.yaml

```diff
@@ -0,0 +1,51 @@
+name: Get ROCKs modified and build-scan-test them
+
+on:
+  workflow_call:
+
+jobs:
+  get-all-rock-paths:
+    name: Get all rockcraft.yaml paths
+    runs-on: ubuntu-22.04
+    outputs:
+      paths: ${{ steps.find.outputs.paths}}
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v3
+      - name: Get all rockcraft.yaml paths using find
+        id: find
+        run: |
+          paths=$(find ./ -name "rockcraft.yaml" -printf "%h:\n  - '%h/**'\n" | sed 's/\.\///g')
+          echo "Paths found:"
+          echo "${paths}"
+          # Passing multi-line string to output requires the use of a delimiter
+          echo "paths<<EOF" >> $GITHUB_OUTPUT
+          echo "${paths}" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+  changes:
```

Comment:
nit: maybe `changed-rocks` or `rock-dirs-with-changes`?  something a little more descriptive?  

If nothing sounds right, `changes` is fine